### PR TITLE
mlp-mixer discriminator

### PR DESCRIPTION
**Pull request by tancredeguillou**

### CODE REVIEW

File: src/evaluation_metrics.py [hunk 1]
```diff
@@ -160,7 +160,7 @@ def read_and_process_image(image_path):
 
 
 def save_evaluation_scores_of_final_images(cfg):
-    if cfg["dataset_type"] == "face":
+    if cfg["dataset_type"] == "face" or cfg["dataset_type"] == "face_reconstruction"):
```
Comment (ardarslan) [inline src/evaluation_metrics.py:163]: For face reconstruction we should report ssim and lpips scores.

File: src/model/discriminator/mlp_mixer.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            strides=[1, self.patch_size, self.patch_size, 1],
+            rates=[1, 1, 1, 1],
+            padding="VALID",
+        )
+        patch_dims = patches.shape[-1]
+        patches = tf.reshape(patches, [batch_size, self.num_patches, patch_dims])
+        return patches
+
+
+def MLPMixerDiscrimator(cfg, positional_encoding=False):
```
Comment (ardarslan) [inline src/model/discriminator/mlp_mixer.py:33]: This should be MLPMixerDiscriminator

File: src/utils.py [hunk 3]
```diff
@@ -163,7 +171,7 @@ def get_model(cfg, model_type):
         elif cfg["discriminator_type"] == "clip":
             ModelClass = CLIPDiscriminator
         elif cfg["discriminator_type"] == "mlp-mixer":
-            raise NotImplementedError()
+            ModelClass = MLPMixerDiscrimator
```
Comment (ardarslan) [inline src/utils.py:174]: This also should be MLPDiscriminator

Comment (ardarslan) [reply src/utils.py:174]: This also should be MLPMixerDiscriminator

**Review by ardarslan — CHANGES_REQUESTED** [review]
I added some comments. Thanks for the PR!

**Review by tancredeguillou — COMMENTED** [review]
Changes updated

**Review by ardarslan — APPROVED** [review]

**Pull request merged**